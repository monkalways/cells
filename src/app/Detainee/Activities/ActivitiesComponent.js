import React from 'react';
import PropTypes from 'prop-types';
import {
  Avatar, Grid, Typography, withStyles,
} from '@material-ui/core';

import MedicalVisitIcon from '../../../images/MedicalVisit.png';
import PhoneAcceptIcon from '../../../images/PhoneAccept.png';
import PhoneDeclineIcon from '../../../images/PhoneDecline.png';
import FingerprintingIcon from '../../../images/Fingerprinting.png';
import BreathalyzerIcon from '../../../images/Breathalyzer.png';
import InterviewIcon from '../../../images/Interview.png';
import BailHearingIcon from '../../../images/BailHearingGeneral.png';

const propTypes = {
  classes: PropTypes.shape({}).isRequired,
  detainee: PropTypes.shape({
    mealAcceptedCount: PropTypes.number.isRequired,
    mealDeclinedCount: PropTypes.number.isRequired,
    medicationAcceptedCount: PropTypes.number.isRequired,
    medicationRefusedCount: PropTypes.number.isRequired,
    verbalCellCheckCount: PropTypes.number.isRequired,
    visualCellCheckCount: PropTypes.number.isRequired,
  }).isRequired,
};

const variant = 'subtitle2';

const ActivitiesComponent = ({ classes, detainee }) => (
  <Grid container className={classes.grid}>
    <Grid item xs={12}>
      <Typography variant="h6" className={classes.text}>
        Activities
      </Typography>
    </Grid>
    <Grid container item xs={12}>
      <Grid item xs={1} className={`${classes.gridItem} ${classes.topLeft}`}>
        <Avatar className={classes.avatar} src={MedicalVisitIcon} />
      </Grid>
      <Grid item xs={1} className={`${classes.gridItem} ${classes.topCenter}`}>
        <Avatar className={classes.avatar} src={PhoneAcceptIcon} />
      </Grid>
      <Grid item xs={1} className={`${classes.gridItem} ${classes.topCenter}`}>
        <Avatar className={classes.avatar} src={PhoneDeclineIcon} />
      </Grid>
      <Grid item xs={1} className={`${classes.gridItem} ${classes.topCenter}`}>
        <Avatar className={classes.avatar} src={FingerprintingIcon} />
      </Grid>
      <Grid item xs={1} className={`${classes.gridItem} ${classes.topCenter}`}>
        <Avatar className={classes.avatar} src={BreathalyzerIcon} />
      </Grid>
      <Grid item xs={1} className={`${classes.gridItem} ${classes.topCenter}`}>
        <Avatar className={classes.avatar} src={InterviewIcon} />
      </Grid>
      <Grid item xs={1} className={`${classes.gridItem} ${classes.topRight}`}>
        <Avatar className={classes.avatar} src={BailHearingIcon} />
      </Grid>
    </Grid>

    <Grid container item xs={12}>
      <Grid item xs={1} className={`${classes.gridItem} ${classes.bottomLeft}`}>
        <Typography variant={variant} align="center">
          {detainee.medicalCount}
        </Typography>
      </Grid>
      <Grid
        item
        xs={1}
        className={`${classes.gridItem} ${classes.bottomCenter}`}
      >
        <Typography variant={variant} align="center">
          {detainee.telephoneAcceptedCount}
        </Typography>
      </Grid>
      <Grid
        item
        xs={1}
        className={`${classes.gridItem} ${classes.bottomCenter}`}
      >
        <Typography variant={variant} align="center">
          {detainee.telephoneDeclinedCount}
        </Typography>
      </Grid>
      <Grid
        item
        xs={1}
        className={`${classes.gridItem} ${classes.bottomCenter}`}
      >
        <Typography variant={variant} align="center">
          {detainee.fingerPrintCount}
        </Typography>
      </Grid>
      <Grid
        item
        xs={1}
        className={`${classes.gridItem} ${classes.bottomCenter}`}
      >
        <Typography variant={variant} align="center">
          {detainee.breathTestCount}
        </Typography>
      </Grid>
      <Grid
        item
        xs={1}
        className={`${classes.gridItem} ${classes.bottomCenter}`}
      >
        <Typography variant={variant} align="center">
          {detainee.interviewCount}
        </Typography>
      </Grid>
      <Grid
        item
        xs={1}
        className={`${classes.gridItem} ${classes.bottomRight}`}
      >
        <Typography variant={variant} align="center">
          {detainee.bailHearingCount}
        </Typography>
      </Grid>
    </Grid>
  </Grid>
);
ActivitiesComponent.propTypes = propTypes;

export default withStyles((theme) => ({
  avatar: {
    height: theme.spacing.unit * 6,
    width: theme.spacing.unit * 6,
  },
  bottomLeft: { borderWidth: '1px 1px 0 0' },
  bottomCenter: { borderWidth: '1px 1px 0 1px' },
  bottomRight: { borderWidth: '1px 0 0 1px' },
  grid: {
    padding: theme.spacing.unit * 0.5,
  },
  gridItem: {
    borderStyle: 'solid',
    borderColor: 'black',
    padding: theme.spacing.unit * 0.5,
  },
  text: {
    textDecoration: 'underline',
  },
  topLeft: { borderWidth: '0 1px 1px 0' },
  topCenter: { borderWidth: '0 1px 1px 1px' },
  topRight: { borderWidth: '0 0 1px 1px' },
}))(ActivitiesComponent);
