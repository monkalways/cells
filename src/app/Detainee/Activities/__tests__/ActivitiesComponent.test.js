import React from 'react';
import { shallow } from 'enzyme';

import { ActivitiesComponent } from '../ActivitiesComponent';

describe('ActivitiesComponent', () => {
  const setup = () => {
    const classes = {
      avatar: 'avatar',
      bottomLeft: 'bottomLeft',
      bottomCenter: 'bottomCenter',
      bottomRight: 'bottomRight',
      grid: 'grid',
      gridItem: 'gridItem',
      text: 'test',
      topLeft: 'topLeft',
      topCenter: 'topCenter',
      topRight: 'topRight',
    };
    return shallow(<ActivitiesComponent
      classes={classes}
      detainee={{
        mealAcceptedCount: 0,
        mealDeclinedCount: 1,
        medicationAcceptedCount: 2,
        medicationRefusedCount: 3,
        verbalCellCheckCount: 4,
        visualCellCheckCount: 5,
      }}
    />);
  };

  it('should render', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
