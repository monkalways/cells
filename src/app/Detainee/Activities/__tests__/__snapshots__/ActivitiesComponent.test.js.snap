// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivitiesComponent should render 1`] = `
<WithStyles(Grid)
  className="grid"
  container={true}
>
  <WithStyles(Grid)
    item={true}
    xs={12}
  >
    <WithStyles(Typography)
      className="test"
      variant="h5"
    >
      Activities
    </WithStyles(Typography)>
  </WithStyles(Grid)>
  <WithStyles(Grid)
    container={true}
    item={true}
    xs={12}
  >
    <WithStyles(Grid)
      className="gridItem topLeft"
      item={true}
    >
      <WithStyles(Avatar)
        className="avatar"
        src="MedicalVisit.png"
      />
    </WithStyles(Grid)>
    <WithStyles(Grid)
      className="gridItem topCenter"
      item={true}
    >
      <WithStyles(Avatar)
        className="avatar"
        src="PhoneAccept.png"
      />
    </WithStyles(Grid)>
    <WithStyles(Grid)
      className="gridItem topCenter"
      item={true}
    >
      <WithStyles(Avatar)
        className="avatar"
        src="PhoneDecline.png"
      />
    </WithStyles(Grid)>
    <WithStyles(Grid)
      className="gridItem topCenter"
      item={true}
    >
      <WithStyles(Avatar)
        className="avatar"
        src="Fingerprinting.png"
      />
    </WithStyles(Grid)>
    <WithStyles(Grid)
      className="gridItem topCenter"
      item={true}
    >
      <WithStyles(Avatar)
        className="avatar"
        src="Breathalyzer.png"
      />
    </WithStyles(Grid)>
    <WithStyles(Grid)
      className="gridItem topCenter"
      item={true}
    >
      <WithStyles(Avatar)
        className="avatar"
        src="Interview.png"
      />
    </WithStyles(Grid)>
    <WithStyles(Grid)
      className="gridItem topRight"
      item={true}
    >
      <WithStyles(Avatar)
        className="avatar"
        src="BailHearingGeneral.png"
      />
    </WithStyles(Grid)>
  </WithStyles(Grid)>
  <WithStyles(Grid)
    container={true}
    item={true}
    xs={12}
  >
    <WithStyles(Grid)
      className="gridItem bottomLeft"
      item={true}
    >
      <WithStyles(Typography)
        align="center"
        variant="h6"
      />
    </WithStyles(Grid)>
    <WithStyles(Grid)
      className="gridItem bottomCenter"
      item={true}
    >
      <WithStyles(Typography)
        align="center"
        variant="h6"
      />
    </WithStyles(Grid)>
    <WithStyles(Grid)
      className="gridItem bottomCenter"
      item={true}
    >
      <WithStyles(Typography)
        align="center"
        variant="h6"
      />
    </WithStyles(Grid)>
    <WithStyles(Grid)
      className="gridItem bottomCenter"
      item={true}
    >
      <WithStyles(Typography)
        align="center"
        variant="h6"
      />
    </WithStyles(Grid)>
    <WithStyles(Grid)
      className="gridItem bottomCenter"
      item={true}
    >
      <WithStyles(Typography)
        align="center"
        variant="h6"
      />
    </WithStyles(Grid)>
    <WithStyles(Grid)
      className="gridItem bottomCenter"
      item={true}
    >
      <WithStyles(Typography)
        align="center"
        variant="h6"
      />
    </WithStyles(Grid)>
    <WithStyles(Grid)
      className="gridItem bottomRight"
      item={true}
    >
      <WithStyles(Typography)
        align="center"
        variant="h6"
      />
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</WithStyles(Grid)>
`;
